packages/backend/src/controllers/authentication/strategies/slackStrategy.ts

```diff
@@ -0,0 +1,71 @@
+/// <reference path="../../../@types/passport-slack-oauth2.d.ts" />
+import {
+    getErrorMessage,
+    OpenIdIdentityIssuerType,
+    OpenIdUser,
+    UnexpectedDatabaseError,
+} from '@lightdash/common';
+import { Strategy as SlackStrategy } from 'passport-slack-oauth2';
+import { lightdashConfig } from '../../../config/lightdashConfig';
+
+export const slackPassportStrategy = !(
+    lightdashConfig.slack?.clientId && lightdashConfig.slack?.clientSecret
+)
+    ? undefined
+    : new SlackStrategy(
+          {
+              scope: ['openid'],
+              clientID: lightdashConfig.slack.clientId,
+              clientSecret: lightdashConfig.slack.clientSecret,
+              callbackURL: `${lightdashConfig.siteUrl.replace(
+                  'http://',
+                  'https://',
+              )}/api/v1/auth/slack/callback`,
+              state: true,
+              passReqToCallback: true,
+          },
+          async (req, accessToken, refreshToken, profile, done) => {
+              if (!req.user || req.user.email === undefined) {
+                  return done(new Error('Invalid user'));
+              }
+
+              const openIdUser: OpenIdUser = {
+                  openId: {
+                      subject: profile.id,
+                      issuer: 'slack-oauth2',
+                      issuerType: OpenIdIdentityIssuerType.SLACK,
+                      email: req.user.email,
+                      firstName: profile.user.name.split(' ')[0],
+                      lastName: profile.user.name.split(' ')[1],
+                      groups: undefined,
+                      teamId: profile.team.id,
+                  },
+              };
+
+              try {
+                  // This might throw an error if the user is already linked to the identity
+                  await req.services
+                      .getUserService()
+                      .linkOpenIdIdentityToUser(req.user, openIdUser);
+                  return done(null);
+              } catch (e: unknown) {
+                  if (
+                      e &&
+                      typeof e === 'object' &&
+                      'code' in e &&
+                      e.code === '23505'
+                  ) {
+                      // Postgres duplicate key error code
+                      console.warn(
+                          `Trying to link open id to user ${req.user.userUuid} but user is already linked`,
+                      );
+                      return done(null); // Silent error if user is already linked to the identity
```

Comment:
you can get this if you try to authenticate multiple times, and we already created the entry in openid. This should not happen in the AI flow,

Reply:
- I logged in from demo@ligtdash.com (user_id = 1)
- then I logged in with demo2@lightdash.com (user_id = 2)

The slack accounts were the same, `user_id` stayed `1`

It’s likely a well-known limitation that we don’t need to address.